### DISCUSSION TITLE
新版本部署后首页根打不开，子目录可以打开

### DISCUSSION THREAD

**Discussion by tkfanswind** [spencerwooo/onedrive-vercel-index#98] [Q&A] [answered]
首页在/Public目录下。 子目录可以打开。  首页报500.估计是bug。报错如下。

`[GET] /api?path=/
21:40:47:31
   _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'GET /v1.0/me/drive/root:%2FPublic%2F:/children?select=%40microsoft.graph.downloadUrl,name,size,id,lastModifiedDateTime,folder,file&top=100 HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJub25jZSI6Il9hcW9WZTZRcGphcmVMc0NMQVgyLWcweHl3YVlhYTRiY0cxWXFfaU1TYkUiLCJhbGciOiJSUzI1NiIsIng1dCI6Im5PbzNaRHJPRFhFSzFqS1doWHNsSFJfS1hFZyIsImtpZCI6Im5PbzNaRHJPRFhFSzFqS1doWHNsSFJfS1hFZyJ9.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTAwMDAtYzAwMC0wMDAwMDAwMDAwMDAiLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC9kNzJkYTgzYS02NjliLTRiMmMtYjBkMC01MjBiZTU4MzgwMzIvIiwiaWF0IjoxNjMwOTM1MzQ3LCJuYmYiOjE2MzA5MzUzNDcsImV4cCI6MTYzMDkzOTI0NywiYWNjdCI6MCwiYWNyIjoiMSIsImFpbyI6IkFTUUEyLzhUQUFBQTB1S2NTdGtSdXRlMzh0cWpiS3JwYzhLMDhOanAzbStmdTJTcmF2TnlmMjQ9IiwiYW1yIjpbInB3ZCJdLCJhcHBfZGlzcGxheW5hbWUiOiJvZHJjIiwiYXBwaWQiOiI5MjY0OGUxOC1iYjViLTQxOGItYjJjMS01NzE3NGUxZTIzOGIiLCJhcHBpZGFjciI6IjEiLCJpZHR5cCI6InVzZXIiLCJpcGFkZHIiOiIxMDMuMTM4LjUzLjMzIiwibmFtZSI6InRrZmFucyIsIm9pZCI6Ijk2MWY2MTUwLTI5N2ItNDBhNS1iZDZjLTU4NDc5MWE0OTViNCIsInBsYXRmIjoiMyIsInB1aWQiOiIxMDAzMjAwMTFBODg1RTNEIiwicmgiOiIwLkFWd0FPcWd0MTV0bUxFdXcwRklMNVlPQU1oaU9aSkpidTR0QnNzRlhGMDRlSTR0Y0FIRS4iLCJzY3AiOiJGaWxlcy5SZWFkIEZpbGVzLlJlYWQuQWxsIEZpbGVzLlJlYWRXcml0ZSBGaWxlcy5SZWFkV3JpdGUuQWxsIFNpdGVzLlJlYWQuQWxsIHByb2ZpbGUgb3BlbmlkIGVtYWlsIiwic3ViIjoiOC05Vkljd3pMMG9ZNlVhLUV5RGw3amtnQ2lRWXhnSkh6UTFaaENKelMzayIsInRlbmFudF9yZWdpb25fc2NvcGUiOiJFVSIsInRpZCI6ImQ3MmRhODNhLTY2OWItNGIyYy1iMGQwLTUyMGJlNTgzODAzMiIsInVuaXF1ZV9uYW1lIjoidGtmYW5zd2luZEBpdC41MjY1MjAueHl6IiwidXBuIjoidGtmYW5zd2luZEBpdC41MjY1MjAueHl6IiwidXRpIjoiLUZDX1NIRGJLVWVGSlRlVVJpX2lBQSIsInZlciI6IjEuMCIsIndpZHMiOlsiYjc5ZmJmNGQtM2VmOS00Njg5LTgxNDMtNzZiMTk0ZTg1NTA5Il0sInhtc19zdCI6eyJzdWIiOiJzMndjMzVJU3RGTWk4dmtkUnRrMkNpQU9udFNObVBKb0NiQmxHVmR0VVEwIn0sInhtc190Y2R0IjoxNTkwNzQ1MTQ1fQ.Psmg4rh62JcodKn7wgoX0D7qe9bqnp4cJ-Z_oTCOehaOWRwzAO91LkzRVkGkZzDZjUPEpxhoNQ9DRTT9nEzHmxPgbZqW7rXkm0HoZpOGy6zEAIy3n85TYwmihpKJfr2uUxMBwY1PQVUEEyhprQDDqC01XYyTgV-6LTvydmzECUw37B_7m8UCtlzAizZYuQKrwsKIWtmUHgmBCKSoyuatMuFp9W-6Hlri56mwe-OQSB9bLx993b7XCBxBf3LJduog9-YiAdTEfi6E7ym-pTSsr_BjjAEV4lAY3r7acr1EuGXLJKWL3qt3vwW05BdECdc6CPMo9cl71O84wICW6huwVA\r\n' +
        'User-Agent: axios/0.21.1\r\n' +
        'Host: graph.microsoft.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/v1.0/me/drive/root:%2FPublic%2F:/children?select=%40microsoft.graph.downloadUrl,name,size,id,lastModifiedDateTime,folder,file&top=100',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'graph.microsoft.com',
      protocol: 'https:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: { error: [Object] }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
RequestId: b0aca11e-e814-4bfc-ad4d-fd3bbb9b90a1 Error: Runtime exited with error: exit status 1
`Runtime.ExitError``

**Comment by zjuyk** [comment]
情况相同，首页 500，但子目录可以访问

**Comment by chen8945** [comment]
俺也一样，我还以为只是我的问题折腾了一下午

**Reply by chen8945** [reply]
> 俺也一样，我还以为只是我的问题折腾了一下午

但是我回滚版本后又能正常使用
只有新版本有这种情况

**Reply by tkfanswind** [reply]
> > 俺也一样，我还以为只是我的问题折腾了一下午
> 
> 但是我回滚版本后又能正常使用
> 只有新版本有这种情况

咋回滚？

**Reply by chen8945** [reply]
> > > 俺也一样，我还以为只是我的问题折腾了一下午
> > 
> > 
> > 但是我回滚版本后又能正常使用
> > 只有新版本有这种情况
> 
> 咋回滚？

git reset --hard commit-id **回滚到commit-id**

**Comment by myl7** [comment, marked as answer]
See #105
我本地测了一下，是解决问题了
感觉 MS 的这个 API 有点离谱

**Reply by 1715173329** [reply]
我测了下似乎没有什么改变，也许是我的问题
主要是在日志里面也只能看到一个 `Error: [Error Object]`，无法得知完整的错误信息

**Reply by myl7** [reply]
`/pages/api/index.ts` 是 MS API 访问的逻辑，或许你可以把里面的 request/response print 出来看看？
我的 fork myl7/onedrive-vercel-index for https://share.myl.moe 是能 work 的，实在需要的话或许可以从我的 fork 改过去？

**Reply by zjuyk** [reply]
实测修复有效，感谢

**Reply by nfway** [reply]
实测无效……

**Reply by myl7** [reply]
> 实测无效……

嗯有 repo 或者是脱敏了的代码的吗？如果有的话我尝试复现一下吧

**Reply by nfway** [reply]
> > 实测无效……
> 
> 嗯有 repo 或者是脱敏了的代码的吗？如果有的话我尝试复现一下吧

用的完全就是你的 repo ，我用导入的形式新开了 private repo
不过使用的是E5

**Reply by myl7** [reply]
> > > 实测无效……
> > 
> > 
> > 嗯有 repo 或者是脱敏了的代码的吗？如果有的话我尝试复现一下吧
> 
> 用的完全就是你的 repo ，我用导入的形式新开了 private repo 不过使用的是E5

其实我的也是 e5
根目录能确认不是对应 / 吗？虽然不确定这有影响，但之前有人报过问题
（如果有精力的话可以尝试把 OneDrive API 的 response console.log 出来脱敏后再给出，这样应该会比较方便其他人排查问题

**Reply by nfway** [reply]
> > > > 实测无效……
> > > 
> > > 
> > > 嗯有 repo 或者是脱敏了的代码的吗？如果有的话我尝试复现一下吧
> > 
> > 
> > 用的完全就是你的 repo ，我用导入的形式新开了 private repo 不过使用的是E5
> 
> 其实我的也是 e5 根目录能确认不是对应 / 吗？虽然不确定这有影响，但之前有人报过问题 （如果有精力的话可以尝试把 OneDrive API 的 response console.log 出来脱敏后再给出，这样应该会比较方便其他人排查问题

realtime  logs 见下

**Reply by nfway** [reply]
> > > > 实测无效……
> > > 
> > > 
> > > 嗯有 repo 或者是脱敏了的代码的吗？如果有的话我尝试复现一下吧
> > 
> > 
> > 用的完全就是你的 repo ，我用导入的形式新开了 private repo 不过使用的是E5
> 
> 其实我的也是 e5 根目录能确认不是对应 / 吗？虽然不确定这有影响，但之前有人报过问题 （如果有精力的话可以尝试把 OneDrive API 的 response console.log 出来脱敏后再给出，这样应该会比较方便其他人排查问题

[GET] /api?path=/
09:15:34:84
xAKk.AgABAAAAAAD--DLA3VO7-xEu5eULRX88NFaa3w89Vzg_kk_CAqDDFPETFBuYxViZ6w3VCTwT8VfCLdDeeLONVTDG54TEDubVoDR4IEYghmlaKbF8nJuREndSzkA9QBov9mTrjY0zJCCLLJYxy5MkZEScNuCN03CcirlDYSjT-CyKoLQQ8sNeqjTUNa4L5iCMh1M_vKV8VWXVZ4liHzc6SkNgjwrHx_9tD-WQF506uuGnrhgfT2GiQvoO4a2CiHTzDcXrgJg7WevrAgDs_wQA9P9T5oAfl0cj4wl2ztjwN_ZDb21iT7wKDQC731olIOwfqRSyy9OvXqsXh_G1F7TGzNSH1gzBdxetVOpmSNwzTIx0ggcOy4ugd7mdJmlFH35blq_W-maqHGyVVY3X2D6q0JPsVQDPDSIAkXD0Y5L68GGTtNdr3R3R5qmnpLlhSaRz3fSz0AysOydCjjruSGeTeUhWqXJ19wX9X_HYaGP205vzcWfhIZS4kbJNI9Q-ye1UaNCT8mcP7eMEs45gRDcFt7Dj9UH3S8lap65Wd1czCxobSAsdy8nJWdhcGXo4D7WzxDOtzg6cSF3vy6uAlOwW9o6mXpIAbrSRQ6AnP4o7AWNtoBPvwM-aFZzGgldQLwrgkvRTWsk9DvoM1dHZBHl05koPqm4Oo12V_u7HzXopS9x50qE0m7sx039PZ54n5pKogshAFc75twMOqd-M-0rgkTSkB96tWaQRW3syrTDhdqwPaw&grant_type=refresh_token',
      headers: [Object],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      adapter: [Function: httpAdapter],
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      validateStatus: [Function: validateStatus]
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: null,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'POST /common/oauth2/v2.0/token HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/x-www-form-urlencoded\r\n' +
        'User-Agent: axios/0.21.1\r\n' +
        'Content-Length: 1213\r\n' +
        'Host: login.microsoftonline.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/common/oauth2/v2.0/token',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'login.microsoftonline.com',
      protocol: 'https:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: {
      error: 'invalid_grant',
      error_description: 'AADSTS700082: The refresh token has expired due to inactivity. The token was issued on 2021-07-15T05:54:14.8651173Z and was inactive for 90.00:00:00.\r\n' +
        'Trace ID: 41994599-8e90-43bf-aaff-340bbab37100\r\n' +
        'Correlation ID: b7f19594-3311-4554-9592-26c1aadce9ac\r\n' +
        'Timestamp: 2021-10-18 01:15:35Z',
      error_codes: [Array],
      timestamp: '2021-10-18 01:15:35Z',
      trace_id: '41994599-8e90-43bf-aaff-340bbab37100',
      correlation_id: 'b7f19594-3311-4554-9592-26c1aadce9ac',
      error_uri: 'https://login.microsoftonline.com/error?code=700082'
    }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
RequestId: da45e62e-807a-4d7a-a08b-83e*****e3 Error: Runtime exited with error: exit status 1
Runtime.ExitError

**Reply by myl7** [reply]
仅从此看的话应该是 `The refresh token has expired due to inactivity`
你试试重新 `npx @beetcb/ms-graph-cli` 生成一下两个 token 试试？
以及如果首页对应的是 / 话也可以换到 /public 之类的试试，不清楚 #124 的问题有没有解决完
（log 全发上来让大家来 debug 也行

**Reply by nfway** [reply]
> 仅从此看的话应该是 `The refresh token has expired due to inactivity` 你试试重新 `npx @beetcb/ms-graph-cli` 生成一下两个 token 试试？ 以及如果首页对应的是 / 话也可以换到 /public 之类的试试，不清楚 #124 的问题有没有解决完 （log 全发上来让大家来 debug 也行

非常感谢，/  与 /public 都不行。
重新生成了 token 之后发现下列问题，但是 client_secret 重新生成了还是出现这问题：

#AYJzdxZA&grant_type=refresh_token', 略
      headers: [Object],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      adapter: [Function: httpAdapter],
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      validateStatus: [Function: validateStatus]
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: null,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'POST /common/oauth2/v2.0/token HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/x-www-form-urlencoded\r\n' +
        'User-Agent: axios/0.21.1\r\n' +
        'Content-Length: 1265\r\n' +
        'Host: login.microsoftonline.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/common/oauth2/v2.0/token',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'login.microsoftonline.com',
      protocol: 'https:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: {
      error: 'invalid_client',
      error_description: 'AADSTS7000215: Invalid client secret is provided.\r\n' +
        'Trace ID: c333958d-9250-4e29-bc57-c83dbaeb0d00\r\n' +
        'Correlation ID: ea26517b-6978-4845-8070-4cc57308cb38\r\n' +
        'Timestamp: 2021-10-18 05:34:22Z',
      error_codes: [Array],
      timestamp: '2021-10-18 05:34:22Z',
      trace_id: 'c333958d-9250-4e29-bc57-c83dbaeb0d00',
      correlation_id: 'ea26517b-6978-4845-8070-4cc57308cb38',
      error_uri: 'https://login.microsoftonline.com/error?code=7000215'
    }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
RequestId: 5e16c623-12e2-4dab-bf96-c4efa4d7701a Error: Runtime exited with error: exit status 1
Runtime.ExitError

**Reply by yueziji** [reply]
> 仅从此看的话应该是 `The refresh token has expired due to inactivity` 你试试重新 `npx @beetcb/ms-graph-cli` 生成一下两个 token 试试？ 以及如果首页对应的是 / 话也可以换到 /public 之类的试试，不清楚 #124 的问题有没有解决完 （log 全发上来让大家来 debug 也行

目前还是未解决的状态，我用的还是老版本😂
